### PR TITLE
[Merged by Bors] - chore(Algebra/Module/Equiv): clean and golf

### PR DESCRIPTION
**Pull request by pitmonticone**

### CODE REVIEW

File: Mathlib/Algebra/Module/Equiv/Basic.lean [hunk 1]
```diff
@@ -122,7 +122,7 @@ protected theorem smul_def (f : M ≃ₗ[R] M) (a : M) : f • a = f a :=
 
 /-- `LinearEquiv.applyDistribMulAction` is faithful. -/
 instance apply_faithfulSMul : FaithfulSMul (M ≃ₗ[R] M) M :=
-  ⟨@fun _ _ => LinearEquiv.ext⟩
+  ⟨@fun _ _ ↦ LinearEquiv.ext⟩
```
Comment (grunweg) [inline Mathlib/Algebra/Module/Equiv/Basic.lean:125]: Does using `fun` also work?

File: Mathlib/Algebra/Module/Equiv/Basic.lean [hunk 2]
```diff
@@ -323,10 +322,10 @@ def addMonoidHomLequivNat {A B : Type*} (R : Type*) [Semiring R] [AddCommMonoid
     where
   toFun := AddMonoidHom.toNatLinearMap
   invFun := LinearMap.toAddMonoidHom
-  map_add' := by intros; ext; rfl
-  map_smul' := by intros; ext; rfl
-  left_inv := by intro f; ext; rfl
-  right_inv := by intro f; ext; rfl
+  map_add' := fun _ _ ↦ rfl
```
Comment (grunweg) [inline Mathlib/Algebra/Module/Equiv/Basic.lean:325]: Does `map_add' _ _ := rfl` work? Similar for the other fields.

File: Mathlib/Algebra/Module/Equiv/Basic.lean [hunk 3]
```diff
@@ -337,17 +336,17 @@ def addMonoidHomLequivInt {A B : Type*} (R : Type*) [Semiring R] [AddCommGroup A
     where
   toFun := AddMonoidHom.toIntLinearMap
   invFun := LinearMap.toAddMonoidHom
-  map_add' := by intros; ext; rfl
-  map_smul' := by intros; ext; rfl
-  left_inv := by intro f; ext; rfl
-  right_inv := by intro f; ext; rfl
+  map_add' := fun _ _ ↦ rfl
```
Comment (grunweg) [inline Mathlib/Algebra/Module/Equiv/Basic.lean:339]: Same as above.

**Review by grunweg — APPROVED** [review]
Thanks! Three minor questions; otherwise looks good to me.

edit: and CI fails... so can you fix that, please?

File: Mathlib/Algebra/Module/Equiv/Basic.lean (hunk 1 above)
Comment (pitmonticone) [reply Mathlib/Algebra/Module/Equiv/Basic.lean:125]: No, it doesn't.

File: Mathlib/Algebra/Module/Equiv/Basic.lean (hunk 2 above)
Comment (pitmonticone) [reply Mathlib/Algebra/Module/Equiv/Basic.lean:325]: Yes, it does. Done, thanks.

File: Mathlib/Algebra/Module/Equiv/Basic.lean (hunk 3 above)
Comment (pitmonticone) [reply Mathlib/Algebra/Module/Equiv/Basic.lean:339]: Ok, done.

**Comment by pitmonticone** [comment]
Thank you @grunweg for your review.

I have merged `master`, so let's see if the checks succeed now. I don't believe the changes I've made could introduce any build errors...

**Comment by pitmonticone** [comment]
Thank you @grunweg for you review! Fixed the problem.

**Comment by grunweg** [comment]
Thanks! This looks good to me now, so let's
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**